docs/services.md

```diff
@@ -266,6 +266,17 @@ In order to allow users to choose a port number for their `Services`, we must
 ensure that no two `Services` can collide.  We do that by allocating each
 `Service` its own IP address.
 
+To ensure each service receives a unique IP, an internal allocator atomically
+updates a global allocation map in etcd prior to each service. The map object
+must exist in the registry for services to get IPs, otherwise creations will
+fail with a message indicating an IP could not be allocated. A background
+controller is responsible for creating that map (to migrate from older versions
+of Kubernetes that used in memory locking) as well as checking for invalid
+assignments due to administrator intervention and cleaning up any any IPs
+that were allocated but which no service currently uses. To alter the portal
+net range, the administrator must delete the service map key and restart the
```

Comment:
To revive the comment, since GitHub obliterated it on a file move:

> > if I understand the model, this creates a single "file" in etcd for serviceipallocations. If we have a port allocator (we will soon) then we duplicate this whole directory tree and change a string constant? Wouldn't we be better to have a top-level /master/ranges dir into which we would write a file called "service-ips" and another called "service-ports" ?

> Maybe. I think eventually you might want to shard to multiple ranges, so absent knowing whether we needed to deal with both single ranges and sharded ranges and per namespace ranges I just tried to get the basics. I prefer having a clear keyspace that deals with ranges (unless you want a recursive delete of a certain keyspace to delete the ranges?).

In a sense this is master config state - maybe we can eventually piggyback on the config state object stuff.  In the mean time, I dread the idea of copying this whole tree for Justin's port allocator.